scripts/sslv2-drown.nse

```diff
@@ -0,0 +1,658 @@
+      on SSLv2 traffic, related to the get_client_master_key and get_client_hello
+      functions.
+    ]],
+    references = {
+      "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2015-3197",
+      "https://www.openssl.org/news/secadv/20160128.txt",
+    },
+  }
+  local cve_2016_0703 = {
+    title = "OpenSSL: Divide-and-conquer session key recovery in SSLv2",
+    state = vulns.STATE.NOT_VULN,
+    IDS = {
+      CVE = 'CVE-2016-0703',
+    },
+    risk_factor = "High",
+    description = [[
+      The get_client_master_key function in s2_srvr.c in the SSLv2 implementation in
+      OpenSSL before 0.9.8zf, 1.0.0 before 1.0.0r, 1.0.1 before 1.0.1m, and 1.0.2 before
+      1.0.2a accepts a nonzero CLIENT-MASTER-KEY CLEAR-KEY-LENGTH value for an arbitrary
+      cipher, which allows man-in-the-middle attackers to determine the MASTER-KEY value
+      and decrypt TLS ciphertext data by leveraging a Bleichenbacher RSA padding oracle, a
+      related issue to CVE-2016-0800.
+    ]],
+    references = {
+      "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0703",
+      "https://www.openssl.org/news/secadv/20160301.txt",
+    },
+  }
+  local cve_2016_0800 = {
+    title = "OpenSSL: Cross-protocol attack on TLS using SSLv2 (DROWN)",
+    state = vulns.STATE.NOT_VULN,
+    IDS = {
+      CVE = 'CVE-2016-0800',
+    },
+    risk_factor = "High",
+    description = [[
+      The SSLv2 protocol, as used in OpenSSL before 1.0.1s and 1.0.2 before 1.0.2g and
+      other products, requires a server to send a ServerVerify message before establishing
+      that a client possesses certain plaintext RSA data, which makes it easier for remote
+      attackers to decrypt TLS ciphertext data by leveraging a Bleichenbacher RSA padding
+      oracle, aka a "DROWN" attack.
+    ]],
+    references = {
+      "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0800",
+      "https://www.openssl.org/news/secadv/20160301.txt",
+    },
+  }
+  report:add_vulns(cve_2015_3197)
+  report:add_vulns(cve_2016_0703)
+  report:add_vulns(cve_2016_0800)
+
+  local offered_ciphers = registry_get(host, port)
+  if not offered_ciphers then -- when the sslv2 was not previously run
+		local sslv2_supported
+		sslv2_supported, offered_ciphers = test_sslv2(host, port)
+		if not sslv2_supported then
+			output.vulns = report:make_output()
+			return output
+		end
+  end
+  output.ciphers = format_ciphers(offered_ciphers)
+
+  -- CVE-2015-3197
+  local forced_ciphers = {}
+  for _, cipher in pairs(SSL_CK) do
+    if try_force_cipher(host, port, cipher) then
+      table.insert(forced_ciphers, cipher)
+    end
+  end
+  output.forced_ciphers = format_ciphers(forced_ciphers)
+  if not values_in(forced_ciphers, offered_ciphers) then
+    cve_2015_3197.state = vulns.STATE.VULN
+  end
+
+  -- CVE-2016-0703
+  for _, cipher in pairs(forced_ciphers) do
+    local result = has_extra_clear_bug(host, port, cipher)
```

Comment:
@bbc2 Do we really need to try this for every discovered cipher? If the bug is there, it should show up no matter what the cipher we choose. I'll test it myself if you don't have further insight.